### ISSUE TITLE
[Firefox] Unable to install: no such file or directory

### ISSUE THREAD

**Issue by crawford** [mozilla/nixpkgs-mozilla#95]
I'm using Nix `2.0.2` on NixOS `18.03.132535.2ee48415afc (Impala)` and trying to install Firefox Beta using `nix-env -iA nixos.latest.firefox-beta-bin`. This results in the following error (full logs below):

    error: getting status of '/nix/store/x087jk2xg3kk93g82i8jcggnp4zfj7mp-firefox-bin-61.0b10/share/icons/hicolor/16x16/apps/firefox.png': No such file or directory

I also get the same error when I attempt to install Nightly or Stable.

The path it's complaining about is a symbolic link to:

    /nix/store/5z75in9cys77bvmrzkxq1sn6b72pg5bl-firefox-release-bin-unwrapped-61.0b10/lib/firefox/browser/chrome/icons/default/default16.png

when it should be:

    /nix/store/5z75in9cys77bvmrzkxq1sn6b72pg5bl-firefox-release-bin-unwrapped-61.0b10/lib/firefox-bin-61.0b10/browser/chrome/icons/default/default16.png

<Details>

```
$ nix-env -iA nixos.latest.firefox-beta-bin
replacing old 'firefox-bin-62.0a1'
installing 'firefox-bin-61.0b10'
these derivations will be built:
  /nix/store/4xfwr5gzdj5w32x3wxy42aprmmvm6954-check-firefox-signature.drv
  /nix/store/9yy04jlnmnxw9d2rxwjnr24jb270h2hm-firefox-61.0b10.tar.bz2.drv
  /nix/store/0zqk7zvcy84pxcdpghylv03iaqhq4spr-firefox-release-bin-unwrapped-61.0b10.drv
  /nix/store/sl7i8iyl218bmqapmkna3rxin661h551-firefox-bin-61.0b10.drv
these paths will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/rws97v54wjp1dib15g1q5ck84zvg4rn3-firefox.desktop
copying path '/nix/store/rws97v54wjp1dib15g1q5ck84zvg4rn3-firefox.desktop' from 'https://cache.nixos.org'...
building '/nix/store/4xfwr5gzdj5w32x3wxy42aprmmvm6954-check-firefox-signature.drv'...
++ cat /nix/store/3bv14sgrx5wp39krx3pk5wcxi8anybv5-firefox.key
++ gpg --import
gpg: directory '/build/tmp.wRFXt8jT5f/.gnupg' created
gpg: keybox '/build/tmp.wRFXt8jT5f/.gnupg/pubring.kbx' created
gpg: key 61B7B526D98F0353: 24 signatures not checked due to missing keys
gpg: /build/tmp.wRFXt8jT5f/.gnupg/trustdb.gpg: trustdb created
gpg: key 61B7B526D98F0353: public key "Mozilla Software Releases <release@mozilla.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: no ultimately trusted keys found
++ gpgv --keyring=/build/tmp.wRFXt8jT5f/.gnupg/pubring.kbx /nix/store/skwy7s6g5dc4rm4yd092a80w90b8mhfm-SHA512SUMS.asc /nix/store/b0vm07rwcffa5
nqnxsldfqvr4if6ws43-SHA512SUMS
gpgv: Signature made Wed May 30 23:52:27 2018 UTC
gpgv:                using RSA key BBBEBDBB24C6F355
gpgv: Good signature from "Mozilla Software Releases <release@mozilla.com>"
++ mkdir /nix/store/s50nf0in58fvljrfr3jy68348a74g9r6-check-firefox-signature
+ eval 'set +o nounset'
++ set +o nounset
+ return
+ exitHandler
+ exitCode=0
+ set +e
+ '[' -n '' ']'
+ ((  0 != 0  ))
+ runHook exitHook
++ shopt -po nounset
+ local 'oldOpts=set +o nounset'
+ set -u
+ local hookName=exitHook
+ shift
+ local 'hooksSlice=exitHooks[@]'
+ local hook
+ for hook in "_callImplicitHook 0 $hookName" ${!hooksSlice+"${!hooksSlice}"}
+ _eval '_callImplicitHook 0 exitHook'
++ type -t '_callImplicitHook 0 exitHook'
+ '[' '' = function ']'
+ set +u
+ eval '_callImplicitHook 0 exitHook'
++ _callImplicitHook 0 exitHook
++ set -u
++ local def=0
++ local hookName=exitHook
++ case "$(type -t "$hookName")" in
+++ type -t exitHook
++ '[' -z '' ']'
++ return 0
+ set -u
+ eval 'set +o nounset'
++ set +o nounset
+ return 0
+ exit 0
building '/nix/store/9yy04jlnmnxw9d2rxwjnr24jb270h2hm-firefox-61.0b10.tar.bz2.drv'...

trying http://download.cdn.mozilla.net/pub/firefox/releases/61.0b10/linux-x86_64/en-US/firefox-61.0b10.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 50.1M  100 50.1M    0     0  2676k      0  0:00:19  0:00:19 --:--:-- 2408k
building '/nix/store/0zqk7zvcy84pxcdpghylv03iaqhq4spr-firefox-release-bin-unwrapped-61.0b10.drv'...
unpacking sources
unpacking source archive /nix/store/xs955dz5rd7kxxacbzbs5aqmzzjs0wgc-firefox-61.0b10.tar.bz2
source root is firefox
setting SOURCE_DATE_EPOCH to timestamp 1527709825 of file firefox/precomplete
installing
post-installation fixup
patching script interpreter paths in /nix/store/5z75in9cys77bvmrzkxq1sn6b72pg5bl-firefox-release-bin-unwrapped-61.0b10
checking for references to /build in /nix/store/5z75in9cys77bvmrzkxq1sn6b72pg5bl-firefox-release-bin-unwrapped-61.0b10...
Wrapping program /nix/store/5z75in9cys77bvmrzkxq1sn6b72pg5bl-firefox-release-bin-unwrapped-61.0b10/bin/firefox
building '/nix/store/sl7i8iyl218bmqapmkna3rxin661h551-firefox-bin-61.0b10.drv'...
building '/nix/store/jyqhc7hwyg9sf6bblrxnl3v9q80jjbdm-user-environment.drv'...
error: getting status of '/nix/store/x087jk2xg3kk93g82i8jcggnp4zfj7mp-firefox-bin-61.0b10/share/icons/hicolor/16x16/apps/firefox.png': No such file or directory
builder for '/nix/store/jyqhc7hwyg9sf6bblrxnl3v9q80jjbdm-user-environment.drv' failed with exit code 1
error: build of '/nix/store/jyqhc7hwyg9sf6bblrxnl3v9q80jjbdm-user-environment.drv' failed
```

**Comment by crawford** [comment]
It looks like this may have been caused by https://github.com/nixos/nixpkgs/commit/01f37880f36e3f0b2f264375ad575062545d0b45.

**Comment by crawford** [comment]
This seems to be fixed on `18.09beta662.b550b7d9e3d (Jellyfish)`.
